droidlet/dialog/dialogue_objects/dialogue_object.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, speaker_name: str, action_dict: Dict, **kwargs):
                     self.speaker_name = speaker_name
                     self.action_dict = action_dict
 
-                def step(self) -> Tuple[Optional[str], Any]:
+                def step(self, agent) -> Tuple[Optional[str], Any]:
```

Comment:
this is just so we can pass it down to respective Tasks or is there more to this ?

Reply:
yes, this is mostly to facilitate the construction of Tasks.  note that previously, the agent was a member of the DialogueObject, so this is a step towards moving agent out (rather than adding it in)